src/data/dfinsupp/lex.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2022 Damiano Testa and Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa, Junyan Xu
+-/
+import data.dfinsupp.order
+import data.dfinsupp.ne_locus
+
+/-!
+# Lexicographic order on finitely supported dependent functions
+This file defines the lexicographic order on `dfinsupp`.
+
+TODO: port `finsupp.lex.linear_order` to `dfinsupp`.
+-/
+
+variables {ι : Type*} {α : ι → Type*}
+
+namespace dfinsupp
+
+section has_zero
+
+variable [Π i, has_zero (α i)]
+
+/-- `dfinsupp.lex r s` is the lexicographic relation on `Π₀ i, α i`, where `ι` is ordered by `r`,
+and `α i` is ordered by `s i`.
+The type synonym `lex (Π₀ i, α i)` has an order given by `dfinsupp.lex (<) (<)`.
```

Comment:
Strictly speaking, this is `dfinsupp.lex (<) (λ _, (<))`; compare the instance below.

Reply:
Thanks; fixed in #16777.